common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerErrorResponse.java

```diff
@@ -0,0 +1,148 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+/**
+ * Encapsulates the broker error result
+ */
+public class BrokerErrorResponse implements Parcelable {
+    private int mErrorCode; //NOPMD
+    private String mErrorMessage;//NOPMD
+    private String mOAuthError;//NOPMD
+    private String mOAuthSubError;//NOPMD
+    private String mOAuthErrorMetadata;//NOPMD
+    private String mOAuthErrorDescription;//NOPMD
+    private String mHttpResponseBody;//NOPMD
+    private String mHttpResponseHeaders;//NOPMD
+
+    public BrokerErrorResponse(){
+
+    }
+
+    protected BrokerErrorResponse(Parcel in) {
+        setErrorCode(in.readInt());
```

Comment:
Should we add `null` check here for `Parcel in`? Otherwise, it will throw `java.lang.NullPointerException`. Same for the other functions.

Reply:
Thanks @heidijinxujia . This should never be the case per documentation but I have added it to be on the safer side